lib/jxl/enc_ans.cc

```diff
@@ -1471,6 +1496,38 @@ void ApplyLZ77(const HistogramParams& params, size_t num_contexts,
 }
 }  // namespace
 
+void EncodeHistograms(const std::vector<uint8_t>& context_map,
+                      const EntropyEncodingData& codes, BitWriter* writer,
+                      size_t layer, AuxOut* aux_out) {
+  BitWriter::Allotment allotment(writer, 128 + kClustersLimit * 136);
+  JXL_CHECK(Bundle::Write(codes.lz77, writer, layer, aux_out));
+  if (codes.lz77.enabled) {
+    EncodeUintConfig(codes.lz77.length_uint_config, writer,
+                     /*log_alpha_size=*/8);
+  }
+  EncodeContextMap(context_map, codes.encoding_info.size(), writer, layer,
+                   aux_out);
+  writer->Write(1, codes.use_prefix_code);
+  size_t log_alpha_size = 8;
+  if (codes.use_prefix_code) {
+    log_alpha_size = PREFIX_MAX_BITS;
+  } else {
+    log_alpha_size = 8;  // streaming_mode
```

Comment:
This comment seems to imply that this function ought to only be used in streaming mode. Is that true?

Reply:
Yes, I added a comment about it in the header.